coffee-backend/src/main/kotlin/chat/letscoffee/user/friend/request/FriendRequestService.kt

```diff
@@ -3,69 +3,64 @@ package chat.letscoffee.user.friend.request
 import chat.letscoffee.exception.BadRequestException
 import chat.letscoffee.exception.ResourceNotFoundException
 import chat.letscoffee.user.User
-import chat.letscoffee.user.UserRepository
+import chat.letscoffee.user.UserService
 import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 
 @Service
-class FriendRequestService(private val repository: FriendRequestRepository, private val userRepository: UserRepository) {
+class FriendRequestService(private val repository: FriendRequestRepository, private val userService: UserService) {
 
-    fun addFriendRequest(from: User, to: String): FriendRequest {
-        val touser: User = userRepository.findByEmail(to)?: throw ResourceNotFoundException("User", "mail", to);
-        // Check if you already have an ongoing request to this user
-
-        if (repository.existsByFromAndTo(from, touser)) {
+    fun addFriendRequest(from: User, to: User): FriendRequest {
+        if (repository.existsByFromAndTo(from, to)) {
             throw BadRequestException("A request between these users already exists")
         }
-        val request = FriendRequest(from = from, to = touser)
+        val request = FriendRequest(from = from, to = to)
 
         return repository.save(request)
     }
 
-    fun getMyFriendRequests(to: User): List<FriendRequest> {
-        return repository.findByToAndAcceptedIsNull(to);
+    fun getRequestsTo(to: User): List<FriendRequest> {
+        return repository.findByToAndAcceptedIsNull(to)
     }
 
 
-    fun getMySentFriendRequests(from: User): List<FriendRequest> {
-        return repository.findByFromAndAcceptedIsNull(from);
+    fun getRequestsFrom(from: User): List<FriendRequest> {
+        return repository.findByFromAndAcceptedIsNull(from)
     }
 
     fun deleteRequest(from: User, id: Long): Boolean {
-        val request: FriendRequest = repository.findByIdOrNull(id)?: throw ResourceNotFoundException("FriendRequest", "id", id) ;
+        val request: FriendRequest = repository.findByIdOrNull(id)?: throw ResourceNotFoundException("FriendRequest", "id", id)
         if (from != request.from) {
-            return false;
+            return false
         }
-        repository.deleteById(id);
+        repository.deleteById(id)
 
-        return true;
+        return true
     }
 
-    fun acceptRequest(to: User, id: Long): Boolean {
-        val request: FriendRequest = repository.findByIdOrNull(id)?: throw ResourceNotFoundException("FriendRequest", "id", id) ;
+    fun acceptRequest(to: User, id: Long) {
+        val request: FriendRequest = repository.findByIdOrNull(id)?: throw ResourceNotFoundException("FriendRequest", "id", id)
         if (to != request.to) {
-            return false;
+            throw ResourceNotFoundException("FriendRequest", "id", id)
```

Comment:
This isn't necessarily right. THe Friend Request exists, you just don't have perissions to do anything with it.

Reply:
Yes, but the user trying to accept a request with an known ID but without knowing which user it is actually for should not be leaked information about it existing. This way it returns 404, the same as if no request could be found (and thus not leaking information)